File: solutions.digamma.damas.jcr/src/main/java/solutions/digamma/damas/jcr/content/JcrDetailedFolder.java
package solutions.digamma.damas.jcr.content;

import solutions.digamma.damas.common.WorkspaceException;
import solutions.digamma.damas.content.DetailedFolder;
import solutions.digamma.damas.inspection.NotNull;

import javax.jcr.Node;

/**
 * Expanded folder.
 *
 * @author Ahmad Shahwan
 */
public class JcrDetailedFolder extends JcrFolder
        implements DetailedFolder, JcrDetailedFile {

    /**
     * Constructor with JCR node.
     *
     * @param node      back-bone JCR node
     */
    public JcrDetailedFolder(@NotNull Node node) throws WorkspaceException {
        super(node);
    }
}

File: solutions.digamma.damas.jcr/src/main/kotlin/solutions/digamma/damas/jcr/content/JcrDetailedFolder.kt
package solutions.digamma.damas.jcr.content

import solutions.digamma.damas.common.WorkspaceException
import solutions.digamma.damas.content.DetailedFolder
import javax.jcr.Node

/**
 * Expanded folder.
 *
 * @param node              back-bone JCR node
 *
 * @author Ahmad Shahwan
 */
internal class JcrDetailedFolder
@Throws(WorkspaceException::class)
private constructor(node: Node) : JcrFolder(node),
        DetailedFolder, JcrDetailedFile {

    companion object {

        @Throws(WorkspaceException::class)
        fun of(node: Node) = JcrDetailedFolder(node)
    }
}
